app/controllers/CheckYourAnswersController.scala

```diff
@@ -44,35 +49,49 @@ class CheckYourAnswersController @Inject()(
   def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
 
-      val list = SummaryListViewModel(
-        rows = Seq(
-          UkimsNumberSummary.row(request.userAnswers),
-          HasNirmsSummary.row(request.userAnswers),
-          NirmsNumberSummary.row(request.userAnswers),
-          HasNiphlSummary.row(request.userAnswers),
-          NiphlNumberSummary.row(request.userAnswers)
-        ).flatten
-      )
+      val (maybeErrors, maybeModel) = TraderProfile.build(request.userAnswers).pad
 
-      Ok(view(list))
+      maybeModel.map { _ =>
+        val list = SummaryListViewModel(
+          rows = Seq(
+            UkimsNumberSummary.row(request.userAnswers),
+            HasNirmsSummary.row(request.userAnswers),
+            NirmsNumberSummary.row(request.userAnswers),
+            HasNiphlSummary.row(request.userAnswers),
+            NiphlNumberSummary.row(request.userAnswers)
+          ).flatten
+        )
+        Ok(view(list))
+      }.getOrElse {
+        logErrorsAndContinue(maybeErrors)
+      }
   }
 
+  def incorrectHasNirms(userAnswers: UserAnswers): Boolean = userAnswers.get(HasNirmsPage).contains(false)  && userAnswers.get(NirmsNumberPage).isDefined
```

Comment:
this isn't being used anywhere

Reply:
This is for discussion with Nick later today - the special cases